ckan/config/config_declaration.yaml

```diff
@@ -1403,6 +1403,111 @@ groups:
         example: 10
         description: The maximum in megabytes an image upload can be.
 
+      - key: ckan.use_classic_uploader
+        type: bool
+        description: |
+            Do not use configurable storages for Uploader and ResourceUploader,
+            keeping previous implementation of these classes. Enable this
+            option to keep backward compatibility with extensions that extend
+            storage logic.
+
+      - key: ckan.files.inline_content_types
+        example: application/pdf image video
+        type: list
+        description: |
+            MIMEtypes that can be served without
+            ``content-disposition:attachment`` header.
+
+      - key: ckan.files.default_storages.default
+        default: default
+        description: |
+          Default storage used for uploads when no explicit storage specified.
+
+      - key: ckan.files.default_storages.admin
+        default: admin_uploads
+        description: |
+          Name of the storage for uploads from AdminUI, such as site logo. The
+          storage must support LINK_PERMANENT capability.
+
+      - key: ckan.files.default_storages.user
+        default: user_uploads
+        description: |
+          Name of the storage for public user uploads, such as user avatar. The
+          storage must support LINK_PERMANENT capability.
+
+      - key: ckan.files.default_storages.group
+        default: group_uploads
+        description: |
+          Name of the storage for public group uploads, such as group
+          image. The storage must support LINK_PERMANENT capability.
+
+      - key: ckan.files.default_storages.resource
+        default: resources
+        description: |
+          Name of the storage for resource uploads. The storage must support
+          STREAM and CREATE capabilities.
+
+      - key: ckan.files.owner.cascade_access
+        validators: default({}) cascade_storage_access
+        default:
+            - package
+            - resource
+            - group
+            - organization
+        example: "package resource:resources group user:avatar"
+        description: |
+          List of owner types that grant access on owned file to anyone who has
+          access to the owner of file. For example, if this option has value
+          ``resource package``, anyone who passes ``resource_show`` auth, can
+          see all files owned by the resource; anyone who passes
+          ``package_show``, can see all files owned by the package; anyone who
+          passes ``package_update``/``resource_update`` can modify files owned
+          by the package/resource; anyone who passes
+          ``package_delete``/``resource_delete`` can delete files owned by the
```

Comment:
A fairly common requirement is to have public metadata (i.e. allow `package_show` or `resource_show`) but restricted downloads. Would be good to have an example on how to do that with the new system (not here, in the docs)